src/zcl_abapgit_folder_logic.clas.abap

```diff
@@ -41,7 +79,11 @@ CLASS ZCL_ABAPGIT_FOLDER_LOGIC IMPLEMENTATION.
     IF iv_top = iv_package.
       rv_path = io_dot->get_starting_folder( ).
     ELSE.
-      lv_parentcl = zcl_abapgit_factory=>get_sap_package( iv_package )->read_parent( ).
+      IF mv_buffered = abap_true.
+        lv_parentcl = zcl_abapgit_factory=>get_sap_package( iv_package )->read_parent( mt_devc_buffer ).
```

Comment:
just read from the local buffer instead of calling the sap_package class? Ideally, this fix does not need any changes to `ZCL_ABAPGIT_SAP_PACKAGE`

Reply:
True, but that would duplicate the parent determination logic, poses a risk if we ever needed to change the logic.
In that case the determination logic would have to be adapted in two places or start to diverge. This way, all the logic to determine sub-/superpackages is collected in `ZCL_ABAPGIT_SAP_PACKAGE`, which feels cleaner and more maintainable to me.

Reply:
Yeah, but the file_status class knows about TDEVC anyhow, if PARENTCL changes there will be a change anyhow,

+      SELECT devclass parentcl
+        FROM tdevc
+        INTO CORRESPONDING FIELDS OF TABLE mt_devc_buffer.

Reply:
Knowing about a list of packages and assuming how they are evaluated/handled are two different things 😉 
If you insist that I change it, I will of course. But I still insist that it would be cleaner and more maintainable code if the functionality how a parent is determined remains separated. 
Consider it and let me know 😃

Reply:
how about having the file_status class buffer the results from the sap_package class? i.e. not doing the initial SELECT, then the logic is completely separated. This would probably suffer a bit of performance, but less memory, and solve potential issues regarding unit testing

Reply:
I understand the unit testing remark. But buffering the results, would end up very close to the original performance, because the performance hit is only partly related to repetition. Most of the improvement comes from the reduced number of single database accesses.
Each DB access is costly (opening the connection, reading, closing th connection) and much more expensive than caluclation. So buffering results will not yield good results I'm afraid.
Maybe it could be a solution to have the folder_logic class inherit from the sap package class or even merge them. That way there would be less interdependency and a clear correlation for any programmer that might adapt the evaluation logic in the future?
What do you think?

Reply:
hmm, the number of objects/files are typically larger than the number of packages, so I'd guess the optimization is mainly related to repetition. Suggest we trying to do some measurements to see how much we are talking about. How many packages do you have in your package structure?

Reply:
Also note TDEVC is fully buffered on the application server, so it is not as expensive as database access, but still slower than having everything buffered in the work process